string[] model_names
---
int32 return_code
